frontend/public/components/default-resource.jsx

```diff
@@ -137,7 +139,12 @@ export const DefaultPage = (props) => (
 DefaultPage.displayName = 'DefaultPage';
 
 export const DefaultDetailsPage = (props) => {
-  const pages = [navFactory.details(DetailsForKind(props.kind)), navFactory.editYaml()];
+  const pages = [
+    navFactory.details(DetailsForKind(props.kind)),
+    navFactory.editYaml(
+      kindForReference(props.kind) === 'TaskRun' ? viewYamlComponent : editYamlComponent,
```

Comment:
This is a generic page. Do not add resource specific handling here. Contribute a new page to override or consider adding a property to models similar to what was done with the badge so that we can provide specific behavior based on model properties.

Reply:
Added a new page for Task Run Overview.